error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/JUnitValueSourceTest.java

```diff
@@ -0,0 +1,414 @@
+            "    return new int[] {0, 1, CONST_INTEGER};",
+            "  }",
+            "",
+            "  @ParameterizedTest",
+            "  @MethodSource(\"intArray\")",
+            "  void boxedInteger(Integer i) {}",
+            "",
+            "  private static LongStream longStream() {",
+            "    return LongStream.of(0, 1, CONST_LONG);",
+            "  }",
+            "",
+            "  @ParameterizedTest",
+            "  @MethodSource(\"longStream\")",
+            "  void primitiveLong(long l) {}",
+            "",
+            "  private static long[] longArray() {",
+            "    return new long[] {0, 1, CONST_LONG};",
+            "  }",
+            "",
+            "  @ParameterizedTest",
+            "  @MethodSource(\"longArray\")",
+            "  void boxedLong(Long l) {}",
+            "",
+            "  private static ImmutableList<Arguments> immutableListOfFloatArguments() {",
+            "    return ImmutableList.of(arguments(0.0F), arguments(1.0F), arguments(CONST_FLOAT));",
+            "  }",
+            "",
+            "  @ParameterizedTest",
+            "  @MethodSource(\"immutableListOfFloatArguments\")",
+            "  void primitiveFloat(float f) {}",
+            "",
+            "  private static Stream<Object> streamOfFloatsAndFloatArguments() {",
+            "    return Stream.of(0.0F, arguments(1.0F), CONST_FLOAT);",
+            "  }",
+            "",
+            "  @ParameterizedTest",
+            "  @MethodSource(\"streamOfFloatsAndFloatArguments\")",
+            "  void boxedFloat(Float f) {}",
+            "",
+            "  private static DoubleStream doubleStream() {",
+            "    return DoubleStream.of(0, 1, CONST_DOUBLE);",
+            "  }",
+            "",
+            "  @ParameterizedTest",
+            "  @MethodSource(\"doubleStream\")",
+            "  void primitiveDouble(double d) {}",
+            "",
+            "  private static double[] doubleArray() {",
+            "    return new double[] {0, 1, CONST_DOUBLE};",
+            "  }",
+            "",
+            "  @ParameterizedTest",
+            "  @MethodSource(\"doubleArray\")",
+            "  void boxedDouble(Double d) {}",
+            "",
+            "  private static ImmutableSet<Arguments> immutableSetOfStringArguments() {",
+            "    return ImmutableSet.of(arguments(\"foo\"), arguments(\"bar\"), arguments(CONST_STRING));",
+            "  }",
+            "",
+            "  @ParameterizedTest",
+            "  @MethodSource(\"immutableSetOfStringArguments\")",
+            "  void string(String s) {}",
+            "",
+            "  private static Stream<Class<?>> streamOfClassesAndClassArguments() {",
+            "    return Stream.of(Stream.class, java.util.Map.class);",
+            "  }",
+            "",
+            "  @ParameterizedTest",
+            "  @MethodSource(\"streamOfClassesAndClassArguments\")",
+            "  void string(Class<?> c) {}",
```

Comment:
```suggestion
            "  void clazz(Class<?> c) {}",
```
(Don't have the code checked out right now; something to fix later.)